src/BlockchainReader/Bitcoin.ts

```diff
@@ -0,0 +1,53 @@
+import { PoetTimestamp, TransactionPoetTimestamp } from '@po.et/poet-js'
+
+import { PREFIX_BARD, PREFIX_POET, Block, Transaction, VOut } from 'Helpers/Bitcoin'
+
+interface VOutWithTxId extends VOut {
+  readonly transactionId: string
+}
+
+export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetTimestamp> =>
+  block.tx
+    .map(transactionToPoetAnchor)
+    .filter(_ => _)
+    .filter(poetAnchorHasCorrectPrefix)
+    .map(poetAnchorWithBlockData(block))
+
+function transactionToPoetAnchor(transaction: Transaction): TransactionPoetTimestamp | undefined {
```

Comment:
This can be written as a pipe() and would avoid the need for all the intermediate variables.
Any reason to use the function keyword vs an arrow function? It would keep it consistent with the other functions in the file.

Reply:
> Any reason to use the function keyword vs an arrow function? It would keep it consistent with the other functions in the file.

I guess I wrote it that way since it has a body and return statement, but either way works fine with me. I'll switch to arrow.

> This can be written as a pipe() and would avoid the need for all the intermediate variables.

I actually started writing it as a pipe originally, but I felt it didn't add as much value as not losing type information, which helps prevent bugs and code faster a great deal. The function is pure either way.

I do agree using pipe leaves less room for bugs (since it removes the need to have a function body and imperative statements completely). Once `pipe` is properly typed I'm definitely switching to it, but for now I feel it adds more value to have the IDE tell me something is wrong in real time for every key stroke and providing autocomplete/intellisense, and the build directly failing if I make a typo.

I'm hoping `pipe` will have proper types soon, I believe TS 3 should support it.

Reply:
Scratch that, `ìdentity` turns everything into `any` until we install `@types/ramda`, but that requires some changes since it causes the build to fail. I'll try to fix things and install `@types/ramda` in a separate PR after mainnet.